azure/scope/machine.go

```diff
@@ -508,7 +508,8 @@ func (m *MachineScope) AvailabilitySetSpec() azure.ResourceSpecGetter {
 func (m *MachineScope) AvailabilitySet() (string, bool) {
 	// AvailabilitySet service is not supported on EdgeZone currently.
 	// AvailabilitySet cannot be used with Spot instances.
-	if !m.AvailabilitySetEnabled() || m.AzureMachine.Spec.SpotVMOptions != nil || m.ExtendedLocation() != nil {
+	if !m.AvailabilitySetEnabled() || m.AzureMachine.Spec.SpotVMOptions != nil || m.ExtendedLocation() != nil ||
```

Comment:
I'm curious why these more specific negative conditions aren't all part of the `AvailabilitySetEnabled()` method?

cc @nojnhuh

Reply:
I think because `AvailabilitySetEnabled()` is currently defined on the Cluster scope where these other machine-specific options aren't available. I could be convinced that we should move it to the Machine scope though where we do have access to both the Cluster and Machine. I wouldn't block this PR on that refactoring though since I see no other uses of `AvailabilitySetEnabled()` besides here.

Reply:
Agree on not blocking, thanks for the pro tip on the scope inheritance.